core/model/job/step/gotolocation/gotolocation.go

```diff
@@ -0,0 +1,105 @@
+package gotolocation
+
+import (
+	"context"
+	"time"
+)
+
+type State int
+type Event int
+
+const (
+	New State = iota
+	Moving
+	Near
+	Arrived
+	Cancelled
+)
+
+func (s State) String() string {
+	return [...]string{"New", "Moving", "Near", "Arrived", "Cancelled"}[s]
+}
+
+const (
+	MoveEvent Event = iota
+	NearEvent
+	ArrivedEvent
+	CancelledEvent
+)
+
+func (e Event) String() string {
+	return [...]string{"MoveEvent", "NearEvent", "ArrivedEvent", "CancelledEvent"}[e]
+}
+
+type GoToLocation struct {
+	DriverID             string  `json:",string"`
+	DestinationLatitude  float64 `json:",string"`
+	DestinationLongitude float64 `json:",string"`
+	UpdatedAt            string  `json:",string"`
+	UpdatedAtLatitude    float64 `json:",string"`
+	UpdatedAtLongitude   float64 `json:",string"`
+	State                State   `json:",string"`
+}
+
+func NewGoToLocation(ctx context.Context, driverID string, latitude, longitude float64) (gtl *GoToLocation, err error) {
+	gtl = &GoToLocation{
+		DriverID:             driverID,
+		DestinationLatitude:  latitude,
+		DestinationLongitude: longitude,
+		UpdatedAt:            time.Now().Format(time.RFC3339Nano),
+		State:                New,
+	}
+
+	err = storage.Store(ctx, *gtl)
+	if err != nil {
+		return
+	}
+	return
+}
+
+func (gtl *GoToLocation) updateState(ctx context.Context, latitude float64, longitude float64, state State) error {
+	gtl.UpdatedAtLatitude = latitude
+	gtl.UpdatedAtLongitude = longitude
+	gtl.UpdatedAt = time.Now().Format(time.RFC3339Nano)
+	gtl.State = state
+
+	err := storage.Store(ctx, *gtl)
```

Comment:
Sorry for the dumb question, but what is `storage` here? A global?

Reply:
its a package global variable, for now, can be wrapped into service or provided/injected in future.
var storage = newStorage(3 * time.Hour)

Why redis, not postgres?

The questions to go through to choose correct storage/repository option for each particular entity:

- Take a closer look on what data we store and what mutations and queries we need to run. set/retrieve by key vs range queries, etc.
- Is this data only makes sense while step is being executed vs the state after execution kept forever for what purpose?
- non-functional requirements.

Redis storage choice in this PR is made on assumptions, subject to change.